activesupport/lib/active_support/callbacks.rb

```diff
@@ -286,9 +286,8 @@ def self.simple(callback_sequence, user_callback)
       class Callback #:nodoc:#
         def self.build(chain, filter, kind, options)
           if filter.is_a?(String)
-            ActiveSupport::Deprecation.warn(<<-MSG.squish)
-              Passing string to define callback is deprecated and will be removed
-              in Rails 5.1 without replacement.
+            raise ArgumentError, <<-MSG.squish
+              Passing string to define callback is not supported. Use proc instead.
             MSG
           end
```

Comment:
We still have string eval code for `if` and `unless` conditional options. Therefore passing string to define callback works again unless raising an error.

https://github.com/rails/rails/blob/v5.0.1/activesupport/test/callbacks_test.rb#L197-L201